projects/cmake/CMakePresets.json

```diff
@@ -1,60 +1,76 @@
-            "binaryDir": "${sourceDir}/build",
-            "cacheVariables": {
-                "CMAKE_C_COMPILER": "gcc",
-                "CMAKE_CXX_COMPILER": "g++",
-                "CMAKE_BUILD_TYPE": "Debug"
-            }
-        },
-        {
-            "name": "macOS",
-            "displayName": "macOS",
-            "description": "macOS clang",
-            "generator": "Unix Makefiles",
-            "binaryDir": "${sourceDir}/build",
-            "cacheVariables": {
-                "CMAKE_C_COMPILER": "clang",
-                "CMAKE_CXX_COMPILER": "clang++",
-                "CMAKE_BUILD_TYPE": "Debug",
-                "USE_COVERAGE": "OFF"
-            }
-        },
-        {
-            "name": "macOSCoverage",
-            "displayName": "macOS with coverage",
-            "description": "macOS clang",
-            "inherits": "macOS",
-            "cacheVariables": {
-                "USE_COVERAGE": "ON"
-            }
+      {
+        "name": "Linux",
+        "displayName": "Linux without coverage",
+        "description": "Linux gcc g++",
+        "binaryDir": "${sourceDir}/build",
+        "cacheVariables": {
+          "CMAKE_C_COMPILER": "gcc",
+          "CMAKE_CXX_COMPILER": "g++",
+          "CMAKE_BUILD_TYPE": "Debug",
+          "USE_COVERAGE": "OFF"
         }
+      },
+      {
+        "name": "LinuxCoverage",
+        "displayName": "Linux with coverage",
+        "description": "Linux gcc g++",
+        "inherits": "Linux",
+        "cacheVariables": {
+          "USE_COVERAGE": "ON"
+        }
+      },
+      {
+        "name": "Windows",
+        "displayName": "Windows",
+        "description": "Windows mingw64 gcc",
+        "generator": "MinGW Makefiles",
+        "binaryDir": "${sourceDir}/build",
+        "cacheVariables": {
+          "CMAKE_C_COMPILER": "gcc",
+          "CMAKE_CXX_COMPILER": "g++",
+          "CMAKE_BUILD_TYPE": "Debug"
+        }
+      },
+      {
+        "name": "macOS",
+        "displayName": "macOS",
+        "description": "macOS clang",
+        "generator": "Unix Makefiles",
+        "binaryDir": "${sourceDir}/build",
+        "cacheVariables": {
+          "CMAKE_C_COMPILER": "clang",
+          "CMAKE_CXX_COMPILER": "clang++",
+          "CMAKE_BUILD_TYPE": "Debug",
+          "USE_COVERAGE": "OFF"
+        }
+      },
+      {
+        "name": "macOSCoverage",
+        "displayName": "macOS Code Coverage",
+        "description": "Configuration for macOS with code coverage enabled",
+        "generator": "Ninja",
+        "binaryDir": "${sourceDir}/build/${presetName}",
+        "cacheVariables": {
+          "CMAKE_BUILD_TYPE": "Debug",
+          "CMAKE_CXX_FLAGS": "--coverage -g -O0",
+          "CMAKE_C_FLAGS": "--coverage -g -O0",
+          "CMAKE_EXE_LINKER_FLAGS": "--coverage"
+        }
+      }
+    ],
+    "buildPresets": [
+      {
+        "name": "macOSCoverage",
+        "configurePreset": "macOSCoverage"
+      }
     ]
```

Comment:
Since this preset isn't changing any properties for the build, I'm not sure it's necessary to add.